### PR TITLE
`pub_covid_hosp_state_timeseries` to use `as_of` parameter

### PR DESCRIPTION
**Pull request by nmdefries**
Closes https://github.com/cmu-delphi/epidatr/issues/209

Note: the server code is [designed to handle `as_of`](https://github.com/cmu-delphi/delphi-epidata/blob/08de322948ac6d55ad394d7c2fab62cbd257372b/src/server/endpoints/covid_hosp_state_timeseries.py#L159) and `issues`, but not `lag`.

### CODE REVIEW

**Review by dshemetov — APPROVED** [review]
lgtm, couple non-blocking suggestions

File: R/endpoints.R [hunk 1]
```diff
@@ -459,30 +459,55 @@ pub_covid_hosp_facility <- function(
...
+  # Check parameters
   rlang::check_dots_empty()
 
+  if (missing(states) || missing(dates)) {
+    stop(
+      "`states` and `dates` are both required"
+    )
+  }
+
+  if (sum(!is.null(issues), !is.null(as_of)) > 1) {
```
[the remark this answers is not in the document]
Comment (dshemetov) [reply R/endpoints.R:491]: there's this snag though: https://github.com/r-lib/rlang/issues/1647

File: R/endpoints.R [hunk 2]
```diff
@@ -459,30 +459,55 @@ pub_covid_hosp_facility <- function(
 #'
 #' @param states character. Two letter state abbreviations.
 #' @param dates [`timeset`]. Dates to fetch.
-#' @param issues [`timeset`]. Optionally, the issues to fetch. If not set, the
-#' most recent issue is returned.
 #' @param ... not used for values, forces later arguments to bind by name
+#' @param as_of Date. Optionally, the as of date for the issues to fetch. If not
+#'   specified, the most recent data is returned. Mutually exclusive with
+#'   `issues` or `lag`.
```
[the remark this answers is not in the document]
Comment (nmdefries) [reply R/endpoints.R:465]: Yeah, there's a lot of opportunity to reduce doc duplication via roxygen templates. I'll make an issue for that.

File: R/endpoints.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmdefries) [reply R/endpoints.R:491]: At the moment, the `sum`-`!is.null` approach is standard across endpoints, so will leave it as-is here.

Opened issue https://github.com/cmu-delphi/epidatr/issues/222

**Pull request merged**